### DISCUSSION TITLE
switchNetwork To HardHat

### DISCUSSION THREAD

**Discussion by mrfujisawa** [wevm/wagmi#353] [Q&A] [answered]
Hey friends,

I wanted to add some code in my app that prompts the user to switch to the local Hardhat network if the app is in dev mode.
I'm using Metamask and I have some code that looks like this - 
```
switchNetwork(id);
```
Where id is `1337` which is the correct `chainId` for my local hardhat ethereum.  When i call the function, MetaMask opens up and correctly asks me if I want to switch to Hardhat.  However, after clicking "switch", MetaMask is still on the same network I had been on previously.  (there was no actual effect). 

Also, there is no error.
I have verified that the underlying request is essentially
```
const id = hexValue(1337);
await window.ethereum!.request({
   method: 'wallet_switchEthereumChain',
   params: [{ chainId: id }],
})
```
This isn't a problem that appears to be specific to Wagmi.  This^ has the same effect.  (no chain in metamask, but prompted to chain network... after clicking switch, Metamask is in the previous state.  and no error is thrown)

Does anyone have any ideas what im doing wrong/how to resolve my problem?

**Comment by tmm** [comment, marked as answer]
Maybe some insight here https://github.com/MetaMask/metamask-extension/issues/10290
